### PR TITLE
Sp24 451 implement game summary page

### PR DESCRIPTION
**Pull request by andreasrberg**
Added new summary game screen:

- Containts speech bubbles for I and O
- High score
- Summary of drawings
- "Play again" and "End game" buttons" which both do the same, as the end game should route to splash screen which does not yet exist

### CODE REVIEW

**Comment by ComputasAlex** [comment]
Hmm

Skjermbilde 2024-07-11 kl  11 15 00

File: src/app/game/speech-bubble/speech-bubble.component.ts
Comment (sondrefj) [inline src/app/game/speech-bubble/speech-bubble.component.ts:1]: Was this the cause for all the logging in console?

File: src/app/game/services/game-state-service.ts [hunk 1]
```diff
@@ -79,6 +80,18 @@ export class GameStateService {
     this.saveGameModeToLocalStorage(GAMEMODE.multiplayer);
   }
 
+  restartGame() {
+    const mode = this._gameMode.value;
+    this._currentPage.next(GAMESTATE.lobby);
+    this.clearState();
+    if (mode === GAMEMODE.singleplayer) {
```
Comment (sondrefj) [inline src/app/game/services/game-state-service.ts:87]:
```suggestion
    if (this.isSinglePlayer()) {
```

File: src/app/game/services/game-state-service.ts [hunk 2]
```diff
@@ -79,6 +80,18 @@ export class GameStateService {
...
   }
 
+  restartGame() {
+    const mode = this._gameMode.value;
+    this._currentPage.next(GAMESTATE.lobby);
+    this.clearState();
+    if (mode === GAMEMODE.singleplayer) {
+      this.setSingleplayer();
+      this.startGame();
+    } else if (mode === GAMEMODE.multiplayer) {
```
Comment (sondrefj) [inline src/app/game/services/game-state-service.ts:90]:
```suggestion
    } else if (this.isMultiplayer()) {

```

**Review by sondrefj — CHANGES_REQUESTED** [review]
Great job! 👏

File: src/app/game/speech-bubble/speech-bubble.component.ts
Comment (andreasrberg) [reply src/app/game/speech-bubble/speech-bubble.component.ts:1]: Yup

**Review by sondrefj — APPROVED** [review]

**Pull request merged**